### PR TITLE
[libc++] Remove _LIBCPP_AUTO_CAST

### PR DESCRIPTION
**Pull request by philnik777**
All supported compilers support `auto()`, so we can drop the workaround. `_LIBCPP_AUTO_CAST(x)` and `auto(x)` are equivalent, but the latter is used in the standard wording, making it a lot easier to verify the code by looking at it. As a bonus, this is probably faster to compile too.

### CODE REVIEW

**Comment by EricWF** [comment]
Even if you think the content of the patch is obvious, please write a description:
Considering including things like:

* Stating the two expressions are equivalent.
* using auto is more obviously correct.
* why the difference between `auto(x)` and `auto{x}` won't make a difference.

**Review by EricWF — CHANGES_REQUESTED** [review]
Please add a description, then otherwise this LGTM.

**Comment by ldionne** [comment]
If no compilers are happy with this change, should we pursue it? If not, let's close.

File: libcxx/include/__ranges/data.h [hunk 1]
```diff
@@ -44,7 +44,7 @@ namespace __data {
     __can_borrow<_Tp> &&
```
Comment (frederick-vs-ja) [inline libcxx/include/__ranges/data.h:44]: `#include <__utility/auto_cast.h>` should also be removed from this file.

**Review by frederick-vs-ja — COMMENTED** [review]
It's strange that `auto(x)` is only erroneous in `<__ranges/access.h>`.

**Comment by philnik777** [comment]
Closing, since it turns out that the auto cast was not back-ported to older standards and we use it in a few places. I'll make a new patch which only removes its uses in C++23-only code.

**Pull request closed without merging**